aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2723,6 +2725,40 @@ static void check_stack_inputs(TensorList tensors, int64_t dim) {
   }
 }
 
+// Pads each tensor on `dim`-th dimension such that padded_dim % num_chunks == 0.
+static std::vector<Tensor> _pad_chunk(TensorList tensors, int64_t dim, int64_t num_chunks) {
+  TORCH_CHECK(!tensors.empty(),
+           "chunk_cat expects a non-empty TensorList");
+  auto num_tensors = tensors.size();
+  std::vector<Tensor> padded_tensors;
+  padded_tensors.reserve(num_tensors);
+  for (const auto & tensor : tensors) {
+    auto tensor_size = tensor.sizes();
+    std::vector<int64_t> padded_size(tensor_size.vec());
+    padded_size[dim] = (tensor_size[dim] + num_chunks - 1) / num_chunks * num_chunks;
```

Comment:
I think we need to check that (1) the number of dims is the same across input tensors (2) `dim` doesn't exceed the number of dims of input tensors.